pokedex/pokemon.lua

```diff
@@ -161,14 +163,24 @@ function M.set_held_item(pokemon, item)
 end
 
 function M.remove_feat(pokemon, feat)
-	for i, name in pairs(M.get_feats(pokemon)) do
-		if name == feat then
-			if name == "Extra Move" then
-				M.remove_move(pokemon, M.get_moves_count(pokemon))
-				table.remove(pokemon.feats, i)
-				break
-			end
-			table.remove(pokemon.feats, i)
+	local removed = 0
+	local feats = M.get_feats(pokemon)
+
+	-- Remove every instance of the feat with this name
```

Comment:
Previously this may not have removed all instances of the feat, depending on how pairs() works (I'm not precisely sure if pairs() lets you remove a thing while iterating it).

Reply:
It would only delete the first "Extra Move" because it `break`s. You can only remove one feat at the time so it shouldn't matter (and `pairs` doesn't like you deleting stuff you iterate over it but as we `break` I was being lazy 🙃 )

Reply:
OK, so I have introduced a new, different bug that every Extra Move feat gets deleted if you delete 1 of them.. is it possible to get more than 1? I would also argue that that bug is less bad than the "totally break the edit screen" bug.

What needs to happen to make this perfect is to remove just 1 Extra Move (ideally the one you clicked the X on but whatever) and then remove the move with index 4+N where N is the number of Extra Move feats you have.

Reply:
We probably don't want to remove all of the feats, the user took an action to remove 1 of the feats. Would feel like a bug if we removed them all. Maybe we should simply remove the "highest" move?
```lua
function M.remove_feat(pokemon, feat)
	local removed = 0
	local feats = M.get_feats(pokemon)

	for i, name in pairs(M.get_feats(pokemon)) do
		if name == feat then
			if name == "Extra Move" then
				M.remove_move(pokemon, M.get_moves_count(pokemon))
				table.remove(pokemon.feats, i)
				break
			end
			table.remove(pokemon.feats, i)
		end
	end

	-- If the feat was extra move, also remove moves with the high indices. There was previously a bug here where if there was a gap in the move set, the move in the last slot would not be removed.
	if feat == "Extra Move" then
		local _, c = M.have_feat(pokemon, "Extra Move")
		M.remove_move(pokemon, M.DEFAULT_MAX_MOVES + c + 1)
	end
end

```

Reply:
Yep that's exactly what I was saying.

Reply:
Oh, that explains a lot. Was a bit confused over your Discord comment. It had not yet showed up when I wrote it for some reason.

Reply:
Fixed! And now it removes the feat you selected rather than the first one / all with that name.